libcaf_core/src/stream_manager.cpp

```diff
@@ -116,15 +113,21 @@ void stream_manager::handle(stream_slots slots, upstream_msg::ack_batch& x) {
 }
 
 void stream_manager::handle(stream_slots slots, upstream_msg::drop&) {
+  CAF_LOG_TRACE(CAF_ARG(slots));
   out().close(slots.receiver);
 }
 
 void stream_manager::handle(stream_slots slots, upstream_msg::forced_drop& x) {
+  CAF_LOG_TRACE(CAF_ARG(slots) << CAF_ARG(x));
   if (out().remove_path(slots.receiver, x.reason, true))
     stop(std::move(x.reason));
 }
 
 void stream_manager::stop(error reason) {
+  CAF_LOG_TRACE(CAF_ARG(reason));
+  if (stopped())
+    return;
+  flags_ = is_stopped_flag;
```

Comment:
`flags_` seems to store bit flags. I'm assuming overwriting them is intended here?

Reply:
Yes, it is. Setting stop clears all other flags.